lisa/tools/iperf3.py

```diff
@@ -47,59 +87,259 @@ def install(self) -> bool:
             self._install_from_src()
         return self._check_exists()
 
-    def run_as_server(self, daemon: bool = True) -> None:
+    def run_as_server_async(
+        self,
+        port: int = 0,
+        format: str = "",
+        interval_seconds: int = 0,
+        use_json_format: bool = False,
+        one_connection_only: bool = False,
+        daemon: bool = True,
+    ) -> Process:
         # -s: run iperf3 as server mode
         # -D: run iperf3 as a daemon
         cmd = " -s"
         if daemon:
-            cmd += " -D"
-        self.run(
-            cmd,
-            force_run=True,
-            sudo=True,
-            shell=True,
+            cmd += " -D "
+        if one_connection_only:
+            cmd += " -1 "
+        if use_json_format:
+            cmd += " -J "
+        if interval_seconds:
+            cmd += f" -i{interval_seconds} "
+        if format:
+            cmd += f" -f {format} "
+        if port:
+            cmd += f" -p {port} "
+        process = self.node.execute_async(
+            f"{self.command} {cmd}", shell=True, sudo=True
+        )
+        return process
+
+    def run_as_server(
+        self,
+        seconds: int = 10,
+        port: int = 0,
+        format: str = "",
+        interval_seconds: int = 0,
+        use_json_format: bool = False,
+        one_connection_only: bool = False,
+        daemon: bool = True,
+    ) -> None:
+        # -s: run iperf3 as server mode
+        # -D: run iperf3 as a daemon
+        # -p: server port to listen on/connect to
+        # -f: [kmgtKMGT] format to report: Kbits, Mbits, Gbits, Tbits
+        # -i: seconds between periodic throughput reports
+        # -1: handle one client connection then exit
+        # -J: output in JSON format
+        process = self.run_as_server_async(
+            port, format, interval_seconds, use_json_format, one_connection_only, daemon
+        )
+        timeout = seconds + 10
```

Comment:
Add comments above to explain it. Try if 2 seconds works. We should save time as much as possible.